### PR TITLE
Voting Assignment

### PR DESCRIPTION
**Pull request by robmadden**

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -30,4 +31,10 @@ def update_rank
     new_rank = points + age_in_days
     update_attribute(:rank, new_rank)
   end
+
+  private
+
+  def create_vote
+    self.user.votes.create(value: 1, post: self)
```
Comment (bmneely) [inline app/models/post.rb]: the first `self` isn't necessary (it also doesn't hurt anything either).

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -51,8 +52,22 @@
...
-        expect( post.points ).to eq(1) # 3 - 2
+        expect( post.points ).to eq(2) # 3 - 2
       end
     end
   end
+
+  describe "#create_vote" do
+    let(:topic) { Topic.create!(name: RandomData.random_sentence, description: RandomData.random_paragraph) }
+    let(:user) { User.create!(name: RandomData.random_name, email: RandomData.random_email, password: "helloworld") }
+    let(:post) { topic.posts.create!(title: RandomData.random_sentence, body: RandomData.random_paragraph, user: user) }
```
Comment (bmneely) [inline spec/models/post_spec.rb]: Let's make this `posts.new` so that we can test the after create callback.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -51,8 +52,22 @@
...
+
+    it "sets the post up_votes value to 1" do
+      expect(post.up_votes).to eq (1)
+    end
+
+    it "sets the post down_votes value to 0" do
+      expect(post.down_votes).to eq (0)
+    end
```
Comment (bmneely) [inline spec/models/post_spec.rb]: How about an extra couple tests:

* A test to ensure that the callback is called.

* A test to ensure that the vote is associated with the owner of the post.

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -5,6 +5,7 @@
   let(:topic) { Topic.create!(name: RandomData.random_sentence, description: RandomData.random_paragraph) }
   let(:user) { User.create!(name: "Bloccit User", email: "user@bloccit.com", password: "helloworld") }
   let(:post) { topic.posts.create!(title: RandomData.random_sentence, body: RandomData.random_paragraph, user: user) }
+  let(:vote) { Vote.create!(value: 1, post: post, user: user) }
```
Comment (bmneely) [inline spec/models/post_spec.rb]: What did we make this change? It doesn't appear to be used in the `#create_vote` specs.

**Comment by bmneely** [comment]
Couple comments, looks good though! There is a merge conflict, but since we're not going to actually mege these branches, that might not matter.

**Comment by robmadden** [comment]
I also noticed that some of the other specs are broken when the new functionality is introduced, is this something we want to address?

**Comment by robmadden** [comment]
Updated specs.

**Comment by bmneely** [comment]
:+1:

**Pull request closed without merging**